app/controllers/passengers_controller.rb

```diff
@@ -14,12 +14,18 @@ def archived
   def brochure; end
 
   def set_status
-    if @passenger.set_status(params[:status])
-      flash[:success] = 'Passenger successfully updated'
+    begin
+      if @passenger.set_status(params[:status])
+        flash[:success] = 'Passenger successfully updated'
+        redirect_to passengers_url
+      else
+        flash[:danger] = @passenger.errors.full_messages
+        redirect_to edit_passenger_path(@passenger)
+      end
+    rescue Net::SMTPFatalError
+      flash[:warning] = 'Passenger successfully updated, ' \
+                        'but the passenger was unable to be notified of their status change via email.'
```

Comment:
A bit too much [passive voice][srbm] for me. Perhaps, "but we could not notify them of their status change" 

[srbm]: https://youtu.be/v-8_LWq236Y